### PR TITLE
Initialize the minimal set of components.

### LINKED ISSUE #16: Crate Namespace
How do we name our crates?  This depends a bit on the answer to #13, but we have two general options:

1. Prefix the names with "icu"
2. Prefix the names with "omnicu"

@zbraniecki had some arguments for preferring "icu".  Can you lay those out here?

### PR DESCRIPTION
**Pull request by zbraniecki**
Fixes #16.

This PR introduces three no-code components with crates in version `0.0.1`.

I'd like to land that, and then add a PR for #31 on top of it.

### CODE REVIEW

File: components/locale/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+description = "API for managing Unicode Langauge and Locale Identifiers"
+version = "0.0.1"
+authors = ["The ICU4X Project Developers"]
+edition = "2018"
+readme = "README.md"
+repository = "https://github.com/unicode-org/omnicu"
+license = "MIT/Apache-2.0"
+categories = ["internationalization"]
+include = [
+    "src/**/*",
```
Comment (nciric) [inline components/locale/Cargo.toml:12]: @Manishearth should we remove src/ part of each package?

File: components/icu4x/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "icu4x"
+description = "International Components for Unicode"
+version = "0.0.1"
+authors = ["The ICU4X Project Developers"]
+edition = "2018"
```
Comment (nciric) [inline components/icu4x/Cargo.toml:6]: Curious, where did 2018 come from?

File: components/locale/Cargo.toml (hunk 1 above)
Comment (zbraniecki) [reply components/locale/Cargo.toml:12]: I'm in favor of keeping `src` directory alongside with `tests`, `benches`, `docs` and `examples` per component.

File: Cargo.toml [hunk 3]
```diff
@@ -1,3 +1,7 @@
 [workspace]
 
-members = []
+members = [
+    "components/icu",
```
Comment (nciric) [inline Cargo.toml:4]: So, icu and icu4x are going to stay dumb placeholders, and locale will have real implementation? We are just defining icu/icu4x here so we can take crates.io namespace.

File: components/icu4x/Cargo.toml (hunk 2 above)
Comment (zbraniecki) [reply components/icu4x/Cargo.toml:6]: Oh boy! Enjoy - https://doc.rust-lang.org/edition-guide/editions/index.html :)

File: Cargo.toml (hunk 3 above)
Comment (zbraniecki) [reply Cargo.toml:4]: Yes. Eventually, I expect `icu` to start operating as an `umbrella` crate that bundles other crates into a single `I want to use ICU4X 1.0`, and `icu4x` will remain a dumb placeholder.

File: components/locale/Cargo.toml (hunk 1 above)
Comment (nciric) [reply components/locale/Cargo.toml:12]: I'll leave you and Manish to duke it out :)

File: components/icu4x/Cargo.toml (hunk 2 above)
Comment (nciric) [reply components/icu4x/Cargo.toml:6]: Ah, so it's a reference to major Rust edition and has nothing to do with licence/start of the project. Good to know.

**Pull request merged**